bitbots_ball_filter/src/bitbots_ball_filter/ball_filter.py

```diff
@@ -107,10 +107,55 @@ def _dynamic_reconfigure_callback(self, config, level):
         self.filter_timer = rospy.Timer(rospy.Duration(self.filter_time_step), self.filter_step)
         return config
 
+    def ball_callback2(self, msg: PoseWithCertaintyArray, optional_distance = True):
```

Comment:
I wouldn't use a second callback. 
Use a ball selection function. Then, we have a select highest rated ball and a select closest to prediction ball function.